notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -34,487 +47,269 @@ import random
-
-#     design_matrix = np.stack(columns, axis=1)
-
-#     var_names = []
-#     for variable in names_dict:
-#         for name in names_dict[variable]:
-#             var_names.append(name)
-
-#     return design_matrix, var_names
-
-
-# def calculate_utilities(betas, design_matrix):
-
-#     limit_max = 700
-#     limit_min = -700
-
-#     utility = design_matrix.dot(betas)
-#     utility[utility > limit_max] = limit_max
-#     utility[utility < limit_min] = limit_min
-
-#     utilities = np.exp(utility)
-
-#     return utilities
-
-
-# def create_mapping_matrix(df_long, observation_id_col):
-#     row_to_col_matrix = pd.get_dummies(df_long[observation_id_col]).values
-# #     row_to_col_matrix = (df_long[observation_id_col].values[:,None] ==
-# #                          np.sort(df_long[observation_id_col].unique())[None,:]).astype(int)
-#     sparse_row_to_col_matrix = sparse.csr_matrix(row_to_col_matrix)
-
-#     mapping_matrix = sparse_row_to_col_matrix.dot(sparse_row_to_col_matrix.T)
-
-#     return mapping_matrix
-
-
-# def calculate_probabilities(betas, design_matrix, mapping_matrix):
-
-#     utilities = calculate_utilities(betas, design_matrix)
-#     denominator = mapping_matrix.dot(utilities)
-#     probabilities = utilities/denominator
-#     probabilities[probabilities == 0] = 1e-300
-
-#     return probabilities
+#import viz
+from causalgraphicalmodels import CausalGraphicalModel, StructuralCausalModel
 ```
 
-## 0.3. Function to simulate choices based on long data format - For Documentation Purposes
+# 0. Rewritten functions (part of refactoring)
 
 ```python
-
-
-# def SimulateChoices(long_data, alt_id_col,
-#                     obs_id_col, number_alts,
-#                     spec_dic, names_dic, init_betas):
-#     """
-#     Function to simulate choices from a long data
-#     format dataset.
-    
-#     Parameters
-#     ----------
-#     long_data : DataFrame
-#         The DataFrame to be used, in long format.
-#     alt_id_col: string
-#         Name of the column containing the alternative
-#         id numbers in the long format dataset.
-#     obs_id_col: string
-#         Name of the column containing the observation
-#         id numbers in the long format dataset.
-#     number_alts: int
-#         Number of alternatives in the long format
-#         dataset.
-#     spec_dic: dictionary
-#         Dictionary of the model specification.
-#     names_dic: dictionary
-#         Dictionary of the alternative names.
-#     init_betas: list
-#         List of the initial betas for the model
-#         from which the choices should be simulated.
+## Functions to replace code within
+## FindLongDataDist
+## Function for checking length
+def isUnique(var_values):
+    return len(var_values.unique()) == 1
+
+## Function for checking if variable is categorical
+def isCategorical(var_type):
+    return var_type == 'categorical'
+
+## Function to get distribution for unique values:
+def getConstantDist(ind_var, var_val):
```

Comment:
I would increase the generality of this function as well as `getCategoricalDist` and `getContinuousDist`.
```suggestion
def getConstantDist(var_name, var_val):
```

Reply:
done!